spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultSchedulerServiceTests.java

```diff
@@ -470,10 +508,20 @@ private ScheduleRequest getScheduleRequest(List<String> commandLineArguments, St
 		TaskDefinition taskDefinition = new TaskDefinition(BASE_DEFINITION_NAME, definition);
 
 		when(mockTaskDefinitionRepository.findById(BASE_DEFINITION_NAME)).thenReturn(Optional.of(taskDefinition));
-		when(mockAppRegistryService.getAppResource(any())).thenReturn(new DockerResource(resourceToReturn));
-		when(mockAppRegistryService.find(taskDefinition.getRegisteredAppName(), ApplicationType.task))
+		doAnswer((Answer<Resource>) invocation -> {
```

Comment:
Another way to do this w/o `doAnswer` is via the `argThat` - something like:
```java
		when(mockAppRegistryService.getAppResource(argThat((appReg) -> DEMO_APP_NAME.equals(appReg.getName()))))
				.thenReturn(new FileSystemResource("file:src/test/resources/apps/foo-task"));

		when(mockAppRegistryService.getAppResource(argThat((appReg) -> !DEMO_APP_NAME.equals(appReg.getName()))))
				.thenReturn(new DockerResource(resourceToReturn));
```

Both work - I try using `argThat` until it breaks down and then I jump into `doAnswer`. Not sure what is best in this case though (about the same IMO ).